build.sbt

```diff
@@ -24,19 +24,75 @@ ThisBuild / crossScalaVersions := Seq(
   Scala3
 ) // There's no reason not to cross-publish
 
-lazy val root = tlCrossRootProject.aggregate(lib)
+lazy val root = tlCrossRootProject.aggregate(core, macros, lib, testkit)
 
+lazy val literallyVersion  = "1.1.0"
 lazy val munitVersion      = "0.7.29"
 lazy val scalacheckVersion = "1.16.0"
 
+lazy val core = crossProject(JVMPlatform, JSPlatform)
+  .crossType(CrossType.Pure)
+  .in(file("core"))
+  .enablePlugins(NoPublishPlugin)
+  .settings(
+    name := "scalac-options-core"
+  )
+
+lazy val macros = crossProject(JVMPlatform, JSPlatform)
+  .crossType(CrossType.Pure)
+  .in(file("macros"))
+  .dependsOn(core)
+  .enablePlugins(NoPublishPlugin)
+  .settings(
+    name := "scalac-options-macros",
+    scalacOptions := {
+      if (tlIsScala3.value)
+        scalacOptions.value.filterNot(_ == "-source:3.0-migration") :+ "-source:3.1"
+      else
+        scalacOptions.value
+    },
+    libraryDependencies ++= Seq(
+      "org.typelevel" %%% "literally" % literallyVersion
```

Comment:
Is there a way for us to check that `literally` doesn't end up on our users' classpath when depending upon this library? Perhaps I'm being paranoid 😛

Reply:
Hmm... I like your idea, but is it feasible in principle? I'll try to figure out that.

Reply:
I don't necessarily mean an automated way to do this but it would be nice to create a project that depends on this library and check the `runtimeClasspath` in sbt 😃